packages/jsii-reflect/lib/tree.ts

```diff
@@ -0,0 +1,278 @@
+import colors = require('colors/safe');
+import { AsciiTree } from 'oo-ascii-tree';
+import { Assembly } from './assembly';
+import { ClassType } from './class';
+import { Dependency } from './dependency';
+import { EnumType } from './enum';
+import { InterfaceType } from './interface';
+import { Method } from './method';
+import { Parameter } from './parameter';
+import { Property } from './property';
+import { TypeReference } from './type-ref';
+import { TypeSystem } from './type-system';
+
+export interface TypeSystemTreeOptions {
+  /**
+   * Show all entity types (supersedes other options)
+   * @default false
+   */
+  showAll?: boolean;
+
+  /**
+   * Show type members (methods, properties)
+   * @default false
+   */
+  members?: boolean;
+
+  /**
+   * Show dependencies
+   * @default false
+   */
+  dependencies?: boolean;
+
+  /**
+   * Show inheritance information (base classes, interfaces)
+   * @default false
+   */
+  inheritance?: boolean;
+
+  /**
+   * Show types
+   * @default false
+   */
+  types?: boolean;
+
+  /**
+   * Show method signatures.
+   * @default false
+   */
+  signatures?: boolean;
+
+  /**
+   * Output with ANSI colors
+   * @default true
+   */
+  colors?: boolean;
+}
+
+/**
+ * Visualizes a `TypeSystem` as an ASCII tree.
+ */
+export class TypeSystemTree extends AsciiTree {
```

Comment:
Why inheritance? Why `is-a` type system tree an ascii tree?

Couldn't `visualizeTypeSystemTree()` be an operation that produces an AsciiTree?

Reply:
It is an is-a relationship, a TypeSystemTree is an AsciiTree. I can rename it to `TypeSystemAsciiTree` if it will be clearer.

I didn't want a dependency between `TypeSystem` and the `AsciiTree`, so those are decoupled class hierarchies. I originally thought to extract `jsii-tree` to a separate module, but for pragmatic reasons decided to just bundle it together with `jsii-reflect`.